### PR TITLE
checkBinary and Converter refactoring

### PR DESCRIPTION
**Pull request by GabrielDonas**
I made the suggested changes above:

- rename misleading variables, input for output
- refactor the checkBinary to return boolean

### CODE REVIEW

File: src/components/converter/Converter.tsx [hunk 1]
```diff
@@ -4,22 +4,22 @@ import checkBinary from "../../helpers/checkBinary";
...
-  const [userInput, setUserInput] = useState("");
+  const [output, setOutput] = useState("");
 
   function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
     const userInputValue = checkBinary(e.target.value);
     if (userInputValue) {
-      const binaryValue = binaryToDecimal(userInputValue);
-      setUserInput(binaryValue);
+      const binaryValue = binaryToDecimal(e.target.value);
+      setOutput(binaryValue);
```
Comment (rypete) [inline src/components/converter/Converter.tsx]: Theres no real need to save `binaryValue` here (other than slight readability). Its `1` space complexity but you could in theory combine these two lines:

```
setOutput(binaryToDecimal(e.target.value));
```

File: src/helpers/checkBinary.ts [hunk 2]
```diff
@@ -1,10 +1,6 @@
 const checkBinary = (value: string) => {
   const isBinary = value.split("").every((n) => n === "0" || n === "1");
-  if (Number(value) < 0 || !isBinary) {
-    return false;
-  } else {
-    return value;
-  }
+  return !(Number(value) < 0 || !isBinary);
```
Comment (rypete) [inline src/helpers/checkBinary.ts]: What is `Number(value)` actually doing for you here? It seems like the statement on line 2 gives you all the validation you need _and_ you pay the time complexity of `n` cost regardless of the outcome of `Number(value)`

Unless I'm missing something you can just return the result of `isBinary`

```
return value.split("").every((n) => n === "0" || n === "1");
```

also the bool flip `!(!isBinary)` is dirty to read

File: src/components/converter/Converter.tsx [hunk 3]
```diff
@@ -4,22 +4,22 @@ import checkBinary from "../../helpers/checkBinary";
 import styles from "./converter.module.css";
 
 function Converter() {
-  const [userInput, setUserInput] = useState("");
+  const [output, setOutput] = useState("");
 
   function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
     const userInputValue = checkBinary(e.target.value);
     if (userInputValue) {
```
Comment (rypete) [inline src/components/converter/Converter.tsx]: Same as the comment below, you don't really need to store this result as you only use it once.

```
if (checkBinary(e.target.value))
```

**Review by rypete — CHANGES_REQUESTED** [review]

File: src/helpers/checkBinary.ts (hunk 2 above)
Comment (GabrielDonas) [reply src/helpers/checkBinary.ts]: That's true. I was doing a redundant check there, in fact, I don't need to check for a positive number. 
Good call.

**Pull request merged**